gst/gst_message_constructors.go

```diff
@@ -8,21 +8,17 @@ import (
 	"github.com/go-gst/go-glib/glib"
 )
 
-func getMessageSourceObj(src interface{}) *C.GstObject {
-	usafe, ok := src.(interface{ Unsafe() unsafe.Pointer })
-	if !ok {
-		return nil
-	}
-	return C.toGstObject(usafe.Unsafe())
+type GstObjectHolder interface {
+	GstObject() *C.GstObject
 }
 
 // NewApplicationMessage creates a new application-typed message. GStreamer will never
 // create these messages; they are a gift from them to you. Enjoy.
 //
 // The source of all message constructors must be a valid Object or descendant, specifically
 // one created from the go runtime. If not the message returned will be nil.
-func NewApplicationMessage(src interface{}, structure *Structure) *Message {
-	srcObj := getMessageSourceObj(src)
+func NewApplicationMessage(src GstObjectHolder, structure *Structure) *Message {
+	srcObj := src.GstObject()
```

Comment:
this is different than before, because the src is not nullable anymore. This will panic on a nullptr. The `getMessageSourceObj` helper function could be changed and used to remove duplicated code, but the nil check needs to be there still:

https://go.dev/play/p/ljn1rU8BM7g

Reply:
Thats true, my bad